src/Plugin/BasePayload.php

```diff
@@ -14,10 +14,14 @@
 use Manticoresearch\Buddy\Core\ManticoreSearch\Endpoint;
 use Manticoresearch\Buddy\Core\ManticoreSearch\Settings;
 use Manticoresearch\Buddy\Core\Network\Request;
+use Manticoresearch\Buddy\Core\Tool\SqlQueryParser;
 
-abstract class BasePayload {
+abstract class BasePayload
+{
 	protected Settings $manticoreSettings;
 
+	public static SqlQueryParser $sqlQueryParser;
```

Comment:
I believe we should avoid exposing the SqlQueryParser instance to external plugins and outside the core. Because it's a wrapper, we need to encapsulate it in such a way that allows us to replace it with another library in the future while maintaining the original response format. If we stick closely to its naming and full interfaces when integrating it outside of our code, we might face significant challenges down the line and potentially break many things.

Reply:
If we could use a simple method in the plugin, like `parse(...)`, that directly gives us results, that would be ideal.

Reply:
Okay, thats great. Juts missed it